src/app/set-password/page.tsx

```diff
@@ -0,0 +1,85 @@
+"use client";
+
+import React, { ReactElement, useEffect, useState } from "react";
+import AuthPanel, { AuthMain } from "@/components/AuthPanel";
+import { updatePassword } from "@/authentication/updatePassword";
+import { useRouter } from "next/navigation";
+import { signInWithTokens } from "@/authentication/signIn";
+import { logErrorReturnLogId, logInfoReturnLogId } from "@/logger/logger";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { AuthError } from "@supabase/gotrue-js";
+
+export default function Page(): ReactElement {
+    const [password, setPassword] = useState("");
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
+    const [accessToken, setAccessToken] = useState<string | null>(null);
+    const [refreshToken, setRefreshToken] = useState<string | null>(null);
+
+    const router = useRouter();
+
+    useEffect(() => {
+        (async () => {
+            const supabase = createClientComponentClient();
```

Comment:
it probably makes more sense to instantiate this just before `const { data } = await supabase.auth.getSession();`